### PR TITLE
Upload Zapbox build to the App Store

### PR DESCRIPTION
**Pull request by mikeskydev**
Automatically build Zapbox in Xcode and upload to App Store.

### CODE REVIEW

File: .github/workflows/build.yml
Comment (mikeage) [inline .github/workflows/build.yml:1071]: Please change this to "iOS App Store", so we don't confuse it with the Mac app store.

File: .github/workflows/build.yml
Comment (mikeage) [inline .github/workflows/build.yml:1112]: this is cute, but what limit?

File: .github/workflows/build.yml
Comment (mikeage) [inline .github/workflows/build.yml:1110]: should all builds go to the app store, or just tags, with regular pushes to main going to Test Flight?

**Review by mikeage — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml
Comment (mikeage) [inline .github/workflows/build.yml:1114]: We use v4 of the upload/download, so you need v4 of this actions

**Comment by mikeage** [comment]
Was this testing E2E in CI? Maybe do one commit having everything except the actual upload (or maybe do the test with TestFlight instead of app store) just to make sure it all runs (if you did, apologies; I'm just not sure how the delete-artifact step could have passed)

**Review by mikeage — DISMISSED** [review]

**Comment by mikeage** [comment]
Incidentally, I think I understand now how to manage developer-id credentials in fastlane too, but I want to wait until this is merged before I refactor #625 to use a git repo instead of secrets.

File: .github/workflows/build.yml
Comment (mikeskydev) [reply .github/workflows/build.yml:1112]: True, this code came from game.ci, so I guess it typically assumes it will be used in closed source!

File: .github/workflows/build.yml
Comment (mikeskydev) [reply .github/workflows/build.yml:1080]: there's also the Gemfile and Gemfile.lock in the root

Comment (mikeage) [reply .github/workflows/build.yml:1080]: OK, good point. You can do three lines, one for each file / directory

**Review by mikeage — COMMENTED** [review]
I'm still curious about the testing, and the app store vs. TestFlight, but if you're satisfied, I'm OK with merging. We can tweak other parts like the cleanup if needed and the checkout optimization later.

File: .github/workflows/build.yml
Comment (mikeage) [reply .github/workflows/build.yml]: FWIW, we might want this to avoid confusion, but I feel like it might still be useful for debugging purposes, at least for a while. 

Maybe later we'll add it to both this intermediate artifact and to the Mac OS (non-DMG) artifact

**Review by mikeage — COMMENTED** [review]
I'm happy enough now not to block, but I can't quite approve yet without confirming testing and TestFlight vs App Store. But if you're happy, don't wait for me ;-)

File: .github/workflows/build.yml
Comment (mikeskydev) [reply .github/workflows/build.yml:1110]: Yeah actually it'd be great to split these two, and honestly do development .ipa on PR builds if possible! I can't work my head around the command line for this, we could obviously triplicate the job but i think we can be a bit more savvy as only this line needs changing. Want to take a shot?

**Comment by mikeage** [comment]
Do we need one to the app store to get this over the line? If so, let's
merge and then change later

File: .github/workflows/build.yml
Comment (mikeage) [reply .github/workflows/build.yml:1110]: Do we need one to the app store to get this over the line? If so, let's
merge and then change later. If not, I can push this change now.

Comment (mikeage) [reply .github/workflows/build.yml:1110]: Actually, ignore this; I want to validate app store flow, and we'll do testflight in another PR

**Pull request merged**
